File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Automation/Alarms/TimeDialogue.java
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel on 3/24/20 1:15 PM
 * Last modified 3/24/20 10:35 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */

package net.geekstools.floatshort.PRO.Automation.Alarms;

import android.app.Activity;
import android.app.TimePickerDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.widget.TimePicker;

import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClass;
import net.geekstools.floatshort.PRO.Utils.Functions.FunctionsClassDebug;

import java.util.Calendar;

public class TimeDialogue extends Activity {

    FunctionsClass functionsClass;

    Calendar newAlarmTime;
    TimePickerDialog timePickerDialog;

    String content, type;
    int position = 0;

    @Override
    protected void onCreate(Bundle saved) {
        super.onCreate(saved);
        functionsClass = new FunctionsClass(getApplicationContext(), this);
        newAlarmTime = Calendar.getInstance();

        content = getIntent().getStringExtra("content");
        type = getIntent().getStringExtra("type");

        timePickerDialog =
                new TimePickerDialog(this, new TimePickerDialog.OnTimeSetListener() {
                    @Override
                    public void onTimeSet(TimePicker timePicker, int hours, int minutes) {
                        newAlarmTime.set(Calendar.HOUR_OF_DAY, hours);
                        newAlarmTime.set(Calendar.MINUTE, minutes);
                        newAlarmTime.set(Calendar.SECOND, 13);

                        String setTime = hours + ":" + minutes;
                        FunctionsClassDebug.Companion.PrintDebug("*** " + setTime);
                        functionsClass.saveFile(
                                content + ".Time",
                                setTime);
                        functionsClass.removeLine(".times.clocks", setTime);
                        functionsClass.saveFileAppendLine(
                                ".times.clocks",
                                setTime);
                        functionsClass.saveFileAppendLine(
                                setTime,
                                content + type);
                        functionsClass.initialAlarm(newAlarmTime, setTime, position);

                        timePickerDialog.dismiss();
                    }
                }, Calendar.getInstance().get(Calendar.HOUR_OF_DAY),
                        Calendar.getInstance().get(Calendar.MINUTE),
                        true);
        timePickerDialog.setCancelable(false);
        timePickerDialog.show();

        timePickerDialog.setOnDismissListener(new DialogInterface.OnDismissListener() {
            @Override
            public void onDismiss(DialogInterface dialog) {
                finish();
            }
        });
    }
}

File: FloatItPhone/src/main/java/net/geekstools/floatshort/PRO/Automation/Alarms/TimeDialogue.kt
/*
 * Copyright © 2020 By Geeks Empire.
 *
 * Created by Elias Fazel
 * Last modified 8/24/20 6:13 AM
 *
 * Licensed Under MIT License.
 * https://opensource.org/licenses/MIT
 */
package net.geekstools.floatshort.PRO.Automation.Alarms

import android.app.Activity
import android.app.TimePickerDialog
import android.app.TimePickerDialog.OnTimeSetListener
import android.os.Bundle
import net.geekstools.floatshort.PRO.Automation.Alarms.Utils.FunctionsClassAlarms
import net.geekstools.floatshort.PRO.Utils.Functions.Debug.Companion.PrintDebug
import net.geekstools.floatshort.PRO.Utils.Functions.FileIO
import java.util.*

class TimeDialogue : Activity() {

    private val fileIO: FileIO by lazy {
        FileIO(applicationContext)
    }

    private lateinit var timePickerDialog: TimePickerDialog

    override fun onCreate(saved: Bundle?) {
        super.onCreate(saved)

        val content: String? = intent.getStringExtra("content")
        val type: String? = intent.getStringExtra("type")

        if (content != null
                && type != null) {

            val functionsClassAlarm = FunctionsClassAlarms(applicationContext)

            val newAlarmTime = Calendar.getInstance()

            timePickerDialog = TimePickerDialog(this@TimeDialogue, OnTimeSetListener { timePicker, hours, minutes ->

                newAlarmTime.set(Calendar.HOUR_OF_DAY, hours)
                newAlarmTime.set(Calendar.MINUTE, minutes)
                newAlarmTime.set(Calendar.SECOND, 13)

                val setTime = "$hours:$minutes"
                PrintDebug("*** $setTime")

                fileIO.saveFile(
                        "$content.Time",
                        setTime)

                fileIO.removeLine(".times.clocks", setTime)
                fileIO.saveFileAppendLine(
                        ".times.clocks",
                        setTime)

                fileIO.saveFileAppendLine(
                        setTime,
                        content + type)

                functionsClassAlarm.initialAlarm(newAlarmTime, setTime)

                timePickerDialog.dismiss()

            }, Calendar.getInstance()[Calendar.HOUR_OF_DAY], Calendar.getInstance()[Calendar.MINUTE], true)

            timePickerDialog.setCancelable(false)
            timePickerDialog.show()

            timePickerDialog.setOnDismissListener {

                this@TimeDialogue.finish()
            }

        }

    }
}